lantern_daemon/tests/daemon_test_with_db.rs

```diff
@@ -285,6 +286,65 @@ async fn test_embedding_generation_runtime(
         .unwrap();
     assert_eq!(cnt, 0);
     assert_eq!(client_data.len(), 0);
+    // Test embedding updates
+    db_client
+        .execute(
+            &format!("UPDATE {CLIENT_TABLE_NAME} SET title=NULL, title_embedding=NULL WHERE id in (1, 2)"),
+            &[],
+        )
+        .await
+        .unwrap();
+    let client_data = db_client
+        .query_one(
+            &format!("SELECT COUNT(*) FROM {CLIENT_TABLE_NAME} WHERE title_embedding IS NULL AND title IS NULL"),
```

Comment:
what are these actually testing?
are you testing somewhere that when you update title, corresponding embedding will be updated by the daemon to the new value?